### PR TITLE
OOP School Library: Setting Up Associations 

### PR DESCRIPTION
**Pull request by aima98**
### For this milestone, I implemented the following features : 

- [ ] Creating a class Classroom
- [ ] Creating the has-many/belongs-to relationship between Classroom and Student. 
- [ ] Creating a class Book
- [ ] Creating a class Rental
- [ ] Creating the many-to-many relationship between Person and Book using the intermediate class Rental. 
- [ ] Creating the has-many side of Book and Rental (a book has many rentals).
- [ ] Creating the belongs-to side of Rental and Book (a rental belongs to a book).
- [ ] Creating the has-many side of Person and Rental (a person has many rentals).
- [ ] Creating the belongs-to side of Rental and Person (a rental belongs to a person).
- [ ] Modifying the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: rentals.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Rental
+  attr_accessor :date, :person, :book
+
+  def initialize(date)
```
Comment (julie-ify) [inline rentals.rb]: - [ ] You are missing the book and person arguments here. Kindly add them.

File: rentals.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    @date = date
+    @person = person
+    @book = book
+
+    # a rental belongs to book
+    @book.add_rental(self)
+
+    # rental belongs to person
+    @person.add_rental(self)
```
Comment (julie-ify) [inline rentals.rb]: - [ ] You are assigning the rental to instance variables (`@book` and `@person`) native to this class. They should be added to the book and person objects (`book` and `person`) that were passed into the constructor.

  ```
  book.add_rental(self)

  person.add_rental(self)
  ```

File: classroom.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class Classroom
+  attr_accessor :label
+  attr_reader :students
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
```
Comment (julie-ify) [inline classroom.rb:10]: - [ ] You did well adding this method for the relationship with the student, but you still need to add a similar method in the Student class.

  ```
  def classroom=(classroom)
    @classroom = classroom
    classroom.students.push(self) unless classroom.students.include?(self)
  end
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @aima98,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by LionRouge1 — APPROVED** [review]
Hi @aima98, 

# STATUS: APPROVED! 🥇 💯 
Great work on making the changes 👏 
You've done well implementing the requested changes

done

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:

## Highlights
- Add setter and getter for `@book` and `@person` :heavy_check_mark: 
- Add `classroom` method to `student` class :heavy_check_mark: 
- All linters passed :heavy_check_mark: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**